Prefer-template multiline

Sometimes we break string in mulitple lines for lisibility like
```js
const a = "that string can get very" +
  `long ${someVar} and we don't want it on one line`
```

The rule warns us to use template string when this is not needing and defeats the purpose.
Right now, there is a work around to add 2 template strings like

```js
const a = `string1` +
  `string2`;
```

But I don't know if this is something we want/should keep. see #3506

Thank you for the report.

I confirmed that.

```js
/* eslint-env es6 */
/* eslint prefer-template:2 */
const a = "that string can get very" +
  `long ${someVar} and we don't want it on one line`;
```

```
> eslint -v
v1.2.1
> eslint test.js --no-eslintrc

test.js
  3:11  error  Unexpected string concatenation  prefer-template

✖ 1 problem (1 error, 0 warnings)
```

I thought this is a false positive.
Two or more string (and template literal) concatenation should be usable to make data without line breaks as keeping readability.

And we should rethink should the next pattern be warned?

```js
let a = "this is a long string" +
    list.map(item => item.name).join(", ") +
    "and more long strings";
```

--- a/lib/rules/prefer-template.js
+++ b/lib/rules/prefer-template.js
@@ -31,7 +31,10 @@ function hasNonLiteral(node) {
     if (node.type === "BinaryExpression") {
         return hasNonLiteral(node.left) || hasNonLiteral(node.right);
     }
-    return node.type !== "Literal";
+    if (node.type === "UnaryExpression") {
+        return hasNonLiteral(node.argument);
+    }
+    return node.type !== "Literal" && node.type !== "TemplateLiteral";
 }
 
 //------------------------------------------------------------------------------

--- a/tests/lib/rules/prefer-template.js
+++ b/tests/lib/rules/prefer-template.js
@@ -28,9 +28,14 @@ ruleTester.run("prefer-template", rule, {
         {code: "'use strict';"},
         {code: "var foo = 'bar';"},
         {code: "var foo = 'bar' + 'baz';"},
+        {code: "var foo = +100 + 'yen';"},
         {code: "var foo = foo + +'100';"},
         {code: "var foo = `bar`;", ecmaFeatures: {templateStrings: true}},
-        {code: "var foo = `hello, ${name}!`;", ecmaFeatures: {templateStrings: true}}
+        {code: "var foo = `hello, ${name}!`;", ecmaFeatures: {templateStrings: true}},
+
+        // https://github.com/eslint/eslint/issues/3507
+        {code: "var foo = `foo` + `bar` + \"hoge\";", ecmaFeatures: {templateStrings: true}},
+        {code: "var foo = `foo` +\n    `bar` +\n    \"hoge\";", ecmaFeatures: {templateStrings: true}}
     ],
     invalid: [
         {code: "var foo = 'hello, ' + name + '!';", errors: errors},

